src/LibraryManager.Vsix/UI/Models/InstallDialogViewModel.cs

```diff
@@ -372,29 +388,186 @@ private static bool IsAnyFileSelected(IReadOnlyList<PackageItem> children)
+
+                        if (containsLibrary)
+                        {
+                            insertionText = _commaDelimiter + Environment.NewLine + newLibrary + Environment.NewLine;
+                        }
+                        else
+                        {
+                            insertionText = newLibrary + Environment.NewLine;
+                        }
+
+                        int insertionIndex = FindInsertionIndex(textBuffer);
+
+                        if (insertionIndex > 0)
+                        {
+                            FormatSelection(textBuffer, insertionIndex, insertionText);
+                        }
+                    }
+                }
+            }
+        }
+
+        private void FormatSelection(ITextBuffer textBuffer, int insertionIndex, string insertionText)
+        {
+            Controller.TextViewData viewData = Controller.TextViewConnectionListener.GetTextViewDataForBuffer(textBuffer);
+            ITextView textView = viewData.LastActiveView;
+
+            CompoundUndoAction compoundUndoAction = new CompoundUndoAction(textView, textBuffer, true);
+            compoundUndoAction.Open(Text.FormatUndoAction);
+
+            using (ITextEdit textEdit = textBuffer.CreateEdit())
+            {
+                textEdit.Insert(insertionIndex, insertionText);
+                textEdit.Apply();
+
+                IOleCommandTarget commandTarget = Shell.Package.GetGlobalService(typeof(Shell.Interop.SUIHostCommandDispatcher)) as IOleCommandTarget;
+
+                if (commandTarget != null)
+                {
+                    SendFocusToEditor(textView);
+
+                    SnapshotSpan snapshotSpan = new SnapshotSpan(textView.TextSnapshot, insertionIndex, insertionText.Length + 1);
+                    textView.Selection.Select(snapshotSpan, false);
+
+                    Guid guidVSStd2K = VSConstants.VSStd2K;
+
+                    commandTarget.Exec(
+                        ref guidVSStd2K,
+                        (uint)VSConstants.VSStd2KCmdID.FORMATSELECTION,
+                        (uint)OLECMDEXECOPT.OLECMDEXECOPT_DODEFAULT,
+                        IntPtr.Zero,
+                        IntPtr.Zero);
+
+                    textView.Selection.Clear();
+                }
+            }
+
+            compoundUndoAction.Close(false);
+        }
+
+        private void SendFocusToEditor(ITextView textView)
+        {
+            FrameworkElement editorWindow = (textView as IWpfTextView).VisualElement;
+            Keyboard.Focus(editorWindow);
+        }
+
+        private JSONMember GetLibraries(ITextBuffer textBuffer)
+        {
+            JSONEditorDocument document = JSONEditorDocument.FromTextBuffer(textBuffer);
+
+            if (document != null)
+            {
+                IEnumerable<JSONMember> jsonMembers = document.JSONDocument.TopLevelValue?.FindType<JSONObject>().Children?.OfType<JSONMember>();
```

Comment:
>. [](start = 113, length = 1)

This doesn't need to be checked?

Reply:
We won't hit this code path if manifest file had syntax errors. E.g: if someone manually deleted the contents of libman.json and then tried to add libraries from UI, we would fail much early in the stack. Let me know if null check here is necessary.

---
In reply to: [197855007](https://github.com/aspnet/LibraryManager/pull/246#discussion_r197855007) [](ancestors = 197855007)

Reply:
OK, as long as you know there is an object in there, then no check is necessary.

---
In reply to: [197922002](https://github.com/aspnet/LibraryManager/pull/246#discussion_r197922002) [](ancestors = 197922002,197855007)